drivers: gpio: Use BIT(n) macro to define GPIO constants

gatzka opened the pull request:
This change would also solve that according to the C11 standard, section 6.5, paragraph 4, the usage of bitwise operators on signed integers is implementation defined.

henrikbrixandersen reviewed (changes requested):
Please squash the two commits, otherwise Git bisect will fail on those format specifiers down the road.

Also, please avoid closing the PR and opening a new one for responding to review comments. Just do the change followed by a force-push.

gatzka committed:
drivers: gpio: Use BIT(n) macro to define GPIO constants

This change would also solve that according to the C11 standard,
section 6.5, paragraph 4, the usage of bitwise operators on
signed integers is implementation defined.

Signed-off-by: Stephan Gatzka <stephan.gatzka@gmail.com>

gatzka commented:
> Please squash the two commits, otherwise Git bisect will fail on those format specifiers down the road.
> 
> Also, please avoid closing the PR and opening a new one for responding to review comments. Just do the change followed by a force-push.

Done.
